rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -2176,6 +2171,31 @@ private <T> T pollQueue() {
         }
     }
 
+    private String getKeyword(String variableName) {
+        String retVal;
+        String leftover = source.substring(cursor);
```

Comment:
We should avoid calling `String#substring()` as that can be very expensive operation for big source files when the parser does this over and over again. One alternative is to use `String#regionMatches()`.